src/tile_cmd.h

```diff
@@ -67,6 +67,7 @@ struct TileDesc {
 	uint16_t road_speed;          ///< Speed limit of road (bridges and track)
 	StringID tramtype;          ///< Type of tram on the tile.
 	uint16_t tram_speed;          ///< Speed limit of tram (bridges and track)
+	std::optional<bool> house_protected; ///< Is this house protected from demolition by towns and AI?
```

Comment:
I wonder if houses the only use for this. The default objects can't be destroyed by anything which could be considered protected?

Reply:
Objects currently have three states: 
1. Can be removed by anything (owned land behavior)
2. Can be removed by normal bulldozer
3. Can only be removed by magic bulldozer.

I don't know if that's worth distinguishing currently, but if #13265 is ever merged (as Sandbox Settings) we could indicate whether objects and industries can be bulldozed. I'll change the names to allow future generic use.